homeassistant/helpers/service.py

```diff
@@ -352,3 +356,40 @@ def async_register_admin_service(
     hass.services.async_register(
         domain, service, admin_handler, schema
     )
+
+
+@bind_hass
+def verify_domain_control(hass: HomeAssistantType, domain: str) -> Callable:
+    """Ensure permission to access any entity under domain in service call."""
+    def decorator(service: 'Service') -> Callable:
+        """Decorate."""
+        @wraps(service)
+        async def check_permissions(
+                call: 'ServiceCall') -> Any:
+            """Check user permission and raise before call if unauthorized."""
+            if not call.context.user_id:
+                return await service(call)
```

Comment:
Code coverage report help me find this line. That should not happen, right? All context should contains user_id, if not, it we should raise exception IMO.

Reply:
If it's an impossible case, I should just remove it.

Reply:
Automations don't have a user ID attached to it.

Reply:
Still don't have? Maybe add owner to automation by default?

Reply:
We should initially limit automation creation to admins while we figure out what we want to do. It gets really tricky because it's hard to validate permissions on what can _trigger_ an automation.